### PR TITLE
#63 feat(Setup) - Set Up Docker Hub and Github Workflow to Push Docker Images

### PR DESCRIPTION
**Pull request by XavierGuertin**
### Summary
Set up a GitHub Actions workflow to automatically build and push Docker images to Docker Hub upon merging pull requests.

### Changes Made
- Added a GitHub Actions workflow to trigger on pushes to the main branch.
- Configured workflow to build Docker images using `docker-compose up --build -d`.
- Integrated Docker Hub login using secure credentials (Docker Hub username and access token).
- Implemented automatic push of built images to Docker Hub after successful builds.
- Added logging and error handling to track build and push issues.

### Testing Instructions
1. Merge a pull request into the main branch.
2. Verify that the GitHub Actions workflow triggers and runs without errors.
3. Check Docker Hub to confirm that the images have been pushed successfully.
4. Inspect logs to confirm the absence of errors in the build and push process.

### Additional Notes
Ensure that the `DOCKER_USERNAME` and `DOCKER_TOKEN` (or access token) secrets are set up correctly in the repository’s settings. Reviewers should confirm that Docker Hub credentials are securely stored and that images are properly tagged in Docker Hub.

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -2,33 +2,43 @@ name: Docker Image CI
...
-
-    - name: Docker Compose Installation
-      run: |
-        sudo apt-get update
-        sudo apt-get install -y docker-compose
-
-    - name: Build the Docker image
-      run: docker-compose up --build -d
-
-    - name: Tear Down the Docker-Compose
```
Comment (LiamDaigle) [inline .github/workflows/docker-build.yml:32]: Why did we get rid of the tear down here?

**Review by LiamDaigle — APPROVED** [review]
lgtm!

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (XavierGuertin) [reply .github/workflows/docker-build.yml:32]: Good point, added it back

**Review by PFratt — APPROVED** [review]
Looks good.

**Pull request merged**